### ISSUE TITLE
[OAuth] Automatically log out/renew if token has expired?

### ISSUE THREAD

**Issue by Wauplin** [gradio-app/gradio#6768]
OAuth scoped tokens currently lives 3600s (cc @coyotte508 to confirm). At the moment, no warning is raised if token has expired which leads to an error if the app uses it. This is annoying if a user logs in a Space, do something and then come back the next day. They will still show as "logged in" even though the token is outdated.

A first solution would be to automatically redirect the user to the "grant permission" page if we detect that the token in their cookie is expired (or at least delete the cookie to show them as "logged out").

Note: it is currently not possible to renew an expired token.

**Comment by Wauplin** [comment]
Yes indeed, we parse it as `gr.OAuthToken.expires_at` attribute in Gradio so should be straightforward to check its value.

**Comment by coyotte508** [comment]
note that now you can configure the oauth expiration (up to 30 days) & there is refresh token support

**Comment by abidlabs** [comment]
Great thanks @coyotte508, the PR checks the actual `expires_at` value from the OAuth session, so regardless of the token expiration, it should behave correctly
